### PR TITLE
Fix Python 3.10 compatibility

### LINKED ISSUE #13: Python 3.10 compatibility
Hi,

It seems `cgroupspy` is not working with python 3.10 with this kind of error:
```
ImportError: cannot import name 'Iterable' from 'collections' (/usr/local/lib/python3.10/collections/__init__.py)
```

As per Python collections documentation, `Iterable` has been moved to `collections.abc` since 3.3 and left for compatibility until 3.9 included: [https://docs.python.org/3.9/library/collections.html](https://docs.python.org/3.9/library/collections.html).

A suggestion for `cgroupspy/interfaces.py`:

Replace
```python
from collections import Iterable
```

By
```python
if sys.version_info.major == 3 and sys.version_info.minor >= 3:
    from collections.abc import Iterable
else:
    from collections import Iterable
```
or equivalent in order to have a distinction between Python 3.3 and older.

### PR DESCRIPTION
**Pull request by potiuk**
Fixes: #13

### CODE REVIEW

**Comment by potiuk** [comment]
@hg-cs @miguel-cs  -> is it possible to merge that single simple change and release a new version of cgroupspy? We are just about to vendor-in cgroups-py in Apache Airlfow : https://github.com/apache/airflow/pull/22209 as this is one of the few last issues that prevent us from providing Python 3.10 compatibility.

**Review by islavov — APPROVED** [review]

File: cgroupspy/interfaces.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 """
-from collections import Iterable
+from collections.abc import Iterable
```
Comment (uranusjr) [inline cgroupspy/interfaces.py:27]:
```suggestion
try:
    from collections.abc import Iterable
except ImportError:
    from collections import Iterable
```

For Python 2 compatibility.

Comment (potiuk) [reply cgroupspy/interfaces.py:27]: Do we have Python 2 support 2 years after End of Life ?

Comment (potiuk) [reply cgroupspy/interfaces.py:27]: I'd rather add Python > 2 in setup.py - especially if there will be no other changes

Comment (potiuk) [reply cgroupspy/interfaces.py:27]: Then Python 2 users will be able to use the current version.

**Comment by potiuk** [comment]
@hg-cs @miguel-cs ?

**Review by anteverse — APPROVED** [review]

**Comment by potiuk** [comment]
Hello?

**Comment by potiuk** [comment]
Anyone?

**Comment by potiuk** [comment]
Helo? @hg-cs @miguel-cs ? Anyone still doing anything here? Should we be considering dumping it as a dependency ?

**Comment by islavov** [comment]
@potiuk original author of the library here. There is a company (cloudsigma) that maintains it, but not sure if it's still a priority for them. Let me try to reach out to them through other channels

**Comment by potiuk** [comment]
Would be fantastic @islavov :). If only to know that we need to look for alternatives. We currently vendored-in the code and fixed it there, but if the library is abandoned, then better to reimplement it :)

**Comment by islavov** [comment]
Reached out, but the person I OOO until mid next week. Will keep you
posted. As for the library, it’s in support only mode. There is no new
development afaik. Seems Miguel is no longer merging pull requests, so
we’ll have to see who took over.

**Comment by potiuk** [comment]
Thanks for reaching out @islavov !

**Comment by jjlopez** [comment]
Hello @potiuk,
First of all, sorry for the delaying response, we are taking the needed actions, so this won't happen again.
About the package, we are still maintaining it. Your pull request will be reviewed and merged in the following days. We will keep you posted.

@islavov, thanks for bringing this issue to our attention.

**Comment by potiuk** [comment]
Thanks @jjlopez

**Comment by cs-cruz** [comment]
Hello @potiuk, sorry for the delay. We have already reviewed your pull request, we made some additional adjustments to keep the compatibility with Python 2.7 which is still in our plans to maintain. The release is already published.

**Comment by potiuk** [comment]
Cool! Thanks. Unvendoring it then :)

**Pull request closed without merging**